src/Sarif.Multitool/RewriteCommand.cs

```diff
@@ -39,6 +41,23 @@ public static int Run(RewriteOptions rewriteOptions)
             return 0;
         }
 
+        private static IDictionary<string, Uri> ConstructUriBaseIds(IEnumerable<string> uriBaseIds)
+        {
+            if (uriBaseIds == null) { return null; }
+
+            IDictionary<string, Uri> result = new Dictionary<string, Uri>();
+
+            foreach (string uriBaseId in uriBaseIds)
+            {
+                string[] tokens = uriBaseId.Split('=');
+                string key = tokens[0];
+                Uri value = new Uri(tokens[1], UriKind.Absolute);
+                result[key] = value;
+            }
```

Comment:
>} [](start = 12, length = 1)

Since the command line parser doesn't enforce the expected `x=y` format, this will crash if there is no `=` (because `tokens[0]` won't exist), or if there's nothing on the right (because `new Uri` will throw, or if there's nothing on the left (because a dictionary can't have a null key).

Reply:
This is a common practice of mine, i don't harden helpers of this kind for specific negative cases. The crash is the current alert to the user. What I usually do is try to put a throttling catch somewhere during the command configuration phase that captures the exception (and reports it), wrapping it as the inner exception of an exception type dedicated to this general condition. This is the way the driver framework operates. Right now, we have lots of inscrutable reports due to missing/improperly constructed arguments.

---
In reply to: [214520418](https://github.com/Microsoft/sarif-sdk/pull/1062#discussion_r214520418) [](ancestors = 214520418)